flink-connector-mongodb/src/test/java/org/apache/flink/connector/mongodb/source/MongoSourceITCase.java

```diff
@@ -0,0 +1,208 @@
+
+    private static final String TEST_DATABASE = "test_source";
+
+    public static final String TEST_COLLECTION = "mongo_source";
+
+    @BeforeAll
+    static void beforeAll() {
+        mongoClient = MongoClients.create(MONGO_CONTAINER.getConnectionString());
+
+        MongoCollection<BsonDocument> coll =
+                mongoClient
+                        .getDatabase(TEST_DATABASE)
+                        .getCollection(TEST_COLLECTION)
+                        .withDocumentClass(BsonDocument.class);
+
+        List<BsonDocument> testRecords = new ArrayList<>();
+        for (int i = 1; i <= 30000; i++) {
+            testRecords.add(createTestData(i));
+            if (testRecords.size() >= 10000) {
+                coll.insertMany(testRecords);
+                testRecords.clear();
+            }
+        }
+    }
+
+    @AfterAll
+    static void afterAll() {
+        if (mongoClient != null) {
+            mongoClient.close();
+        }
+    }
+
+    @ParameterizedTest
+    @EnumSource(
+            value = PartitionStrategy.class,
+            names = {"SHARDED"},
+            mode = EXCLUDE)
+    public void testPartitionStrategy(PartitionStrategy partitionStrategy) throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(2);
+
+        MongoSource<RowData> mongoSource =
+                defaultSourceBuilder()
+                        .setPartitionSize(MemorySize.parse("1mb"))
+                        .setSamplesPerPartition(3)
+                        .setPartitionStrategy(partitionStrategy)
+                        .build();
+
+        List<RowData> results =
+                CollectionUtil.iteratorToList(
+                        env.fromSource(
+                                        mongoSource,
+                                        WatermarkStrategy.noWatermarks(),
+                                        "MongoDB-Source")
+                                .executeAndCollect());
+
+        assertThat(results).hasSize(30000);
+    }
+
+    @Test
+    public void testLimit() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(2);
+
+        MongoSource<RowData> mongoSource = defaultSourceBuilder().setLimit(100).build();
+
+        List<RowData> results =
+                CollectionUtil.iteratorToList(
+                        env.fromSource(
+                                        mongoSource,
+                                        WatermarkStrategy.noWatermarks(),
+                                        "MongoDB-Source")
+                                .executeAndCollect());
+
+        assertThat(results).hasSize(100);
+    }
+
+    @Test
+    public void testProject() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(2);
```

Comment:
This shouldn't be necessary. The mini cluster automatically defaults to the maximum parallelism that is can support with the given number of tms and slots_per_tm.